src/data/client.js

```diff
@@ -64,7 +64,24 @@ export const uploadDataset = async (s3Params, file, request, confirmUpload) => {
     dataset.key = response.s3Data.key;
     dataset.submitted_at = new Date();
     // console.log(JSON.stringify(dataset))
-    confirmUpload(true);
+    const uploadResponse = await fetch('api/datasets', 
+      {
+        method: "POST",
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(dataset)
+      }
+    );
+    
+    if (uploadResponse.ok)
+    {
+      confirmUpload(true);
+    }
+    else
+    {
+      confirmUpload(false);
```

Comment:
Is this gonna work? You're passing `confirmUpload` as a callback here but I couldn't find the source and I have no idea what will happen if it gets called with false instead of true. @meeeech